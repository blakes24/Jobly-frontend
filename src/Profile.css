.Profile .card-title {
  font-size: 2rem;
  text-align: center;
  margin-top: 1rem;
  margin-bottom: 0;
}

.Profile .card {
  -webkit-box-shadow: 5px 5px 10px 2px lightgrey;
  box-shadow: 3px 3px 10px 2px lightgrey;
  max-width: 500px;
  margin: 0 auto;
  margin-top: 15vh;
}

.Profile form {
  padding: 1rem;
}

.Profile .form-label {
  margin: 10px 0 5px;
}

.Profile .btn {
  margin-top: .3rem;
}

.Profile-message {
  color: #00992e;
  margin-top: .5rem;
  text-align: center;
  min-height: 16px;
  line-height: 16px;
}

@media screen and (max-width: 600px) {
  .Profile .card {
    margin-top: .5rem;
    box-shadow: none;
    border: none;
  }
  .Profile .card-body {
    background-color: #f1f2f3;
    font-weight: 500;
  }
  .Profile .card-title {
    margin-top: 0;
  }
}
