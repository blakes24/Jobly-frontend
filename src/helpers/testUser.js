import UserContext from './UserContext';

const testUser = {
  username     : 'testuser',
  firstName    : 'Test',
  lastName     : 'User',
  email        : 'joel@joelburton.com',
  isAdmin      : false,
  applications : []
};

const UserProvider = ({ children, user = testUser }) => (
  <UserContext.Provider value={{ user }}>{children}</UserContext.Provider>
);

export { UserProvider };
