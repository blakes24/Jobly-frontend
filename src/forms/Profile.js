import React, { useContext } from 'react';
import { useFormik } from 'formik';
import { Form, Button, Card } from 'react-bootstrap';
import UserContext from '../helpers/UserContext';
import './Profile.css';

const validate = (values) => {
  const errors = {};

  if (!values.firstName) {
    errors.firstName = 'Required';
  } else if (values.firstName.length > 15) {
    errors.firstName = 'Must be 15 characters or less';
  }

  if (!values.lastName) {
    errors.lastName = 'Required';
  } else if (values.lastName.length > 20) {
    errors.lastName = 'Must be 20 characters or less';
  }

  if (!values.email) {
    errors.email = 'Required';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Invalid email address';
  }

  if (!values.password) {
    errors.password = 'Required';
  }

  return errors;
};

const Profile = ({ update }) => {
  const { user } = useContext(UserContext);

  const formik = useFormik({
    initialValues : {
      username  : user.username,
      firstName : user.firstName,
      lastName  : user.lastName,
      email     : user.email,
      password  : ''
    },
    validate,
    onSubmit      : async (values, actions) => {
      try {
        await update(values);
        actions.setStatus({ message: 'Profile Updated' });
      } catch (err) {
        formik.errors.password = err;
      }
    }
  });
  return (
    <div className="Profile">
      <Card>
        <Card.Body>
          <Card.Title>Update Profile</Card.Title>
          <Form onSubmit={formik.handleSubmit}>
            <Form.Label htmlFor="username">Username</Form.Label>
            <Form.Control
              plaintext
              readOnly
              className="text-secondary"
              id="username"
              name="username"
              type="text"
              defaultValue={formik.values.username}
            />
            {formik.touched.username && formik.errors.username ? (
              <Form.Text className="text-danger">{formik.errors.username}</Form.Text>
            ) : null}

            <Form.Label htmlFor="firstName">First Name</Form.Label>
            <Form.Control
              id="firstName"
              name="firstName"
              type="text"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              value={formik.values.firstName}
            />
            {formik.touched.firstName && formik.errors.firstName ? (
              <Form.Text className="text-danger">{formik.errors.firstName}</Form.Text>
            ) : null}

            <Form.Label htmlFor="lastName">Last Name</Form.Label>
            <Form.Control
              id="lastName"
              name="lastName"
              type="text"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              value={formik.values.lastName}
            />
            {formik.touched.lastName && formik.errors.lastName ? (
              <Form.Text className="text-danger">{formik.errors.lastName}</Form.Text>
            ) : null}

            <Form.Label htmlFor="email">Email Address</Form.Label>
            <Form.Control
              id="email"
              name="email"
              type="email"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              value={formik.values.email}
            />
            {formik.touched.email && formik.errors.email ? (
              <Form.Text className="text-danger">{formik.errors.email}</Form.Text>
            ) : null}

            <Form.Label htmlFor="password">Confirm password to make changes</Form.Label>
            <Form.Control
              id="password"
              name="password"
              type="password"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              value={formik.values.password}
            />
            {formik.touched.password && formik.errors.password ? (
              <Form.Text className="text-danger">{formik.errors.password}</Form.Text>
            ) : null}

            <div className="Profile-message">
              {formik.status && formik.status.message ? formik.status.message : ' '}
            </div>

            <Button type="submit">Submit</Button>
          </Form>
        </Card.Body>
      </Card>
    </div>
  );
};

export default Profile;
