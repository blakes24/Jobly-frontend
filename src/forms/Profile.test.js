import { render } from '@testing-library/react';
import { UserProvider } from '../helpers/testUser';
import Profile from './Profile';

test('renders without crashing', () => {
  render(
    <UserProvider>
      <Profile />
    </UserProvider>
  );
});

it('matches snapshot', function() {
  const { asFragment } = render(
    <UserProvider>
      <Profile />
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
