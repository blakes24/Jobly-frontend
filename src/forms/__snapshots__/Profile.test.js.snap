// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Profile"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          Update Profile
        </div>
        <form
          class=""
        >
          <label
            class="form-label"
            for="username"
          >
            Username
          </label>
          <input
            class="text-secondary form-control-plaintext"
            id="username"
            name="username"
            readonly=""
            type="text"
            value="testuser"
          />
          <label
            class="form-label"
            for="firstName"
          >
            First Name
          </label>
          <input
            class="form-control"
            id="firstName"
            name="firstName"
            type="text"
            value="Test"
          />
          <label
            class="form-label"
            for="lastName"
          >
            Last Name
          </label>
          <input
            class="form-control"
            id="lastName"
            name="lastName"
            type="text"
            value="User"
          />
          <label
            class="form-label"
            for="email"
          >
            Email Address
          </label>
          <input
            class="form-control"
            id="email"
            name="email"
            type="email"
            value="joel@joelburton.com"
          />
          <label
            class="form-label"
            for="password"
          >
            Confirm password to make changes
          </label>
          <input
            class="form-control"
            id="password"
            name="password"
            type="password"
            value=""
          />
          <div
            class="Profile-message"
          >
             
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
